app/models/field/embed.rb

```diff
@@ -0,0 +1,108 @@
+require 'uri'
+
+class Field::Embed < ::Field
+  store_accessor :options, :format
+  store_accessor :options, :width
+  store_accessor :options, :height
+  store_accessor :options, :domains
+  after_save :remove_width_height_value, :if => :code?
+
+  FORMATS = %w(url code).freeze
+
+  def item_link
+    "/#{@item.catalog.slug}/#{I18n.locale}/#{@item.item_type.slug}/#{@item.id}"
+  end
+
+  def sql_type
+    "VARCHAR(512)"
+  end
+
+  def url?
+    options && options['format'] && options['format'] == 'url'
```

Comment:
```suggestion
    options && options['format'] == 'url'
```

(pareil pour `def code?`)

Reply:
d'ailleurs pourquoi ne pas directement lire la valeur `format`, car le `store_accessor` est justement la pour en faire une méthode :
```suggestion
    format == 'url'
```
(pareil pour tous les autres accessors)

Reply:
replaced with accessors